app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java

```diff
@@ -224,72 +202,66 @@ private static MetricsQuery useAvailableRollups(
+                                .stream()
+                                .filter(IS_ROLLUP)
+                                .filter(s -> s.length() == metricName.length() + 3)
+                                .collect(Collectors.toList());
+
+                        if (rollupMetrics.isEmpty()) {
+                            metrics.incrementCounter("kairosService/useRollups/noRollups", 1);
+                            // No rollups so execute what we received
+                            return metric;
+                        } else {
+                            // There are rollups, now determine the appropriate one based on the max sampling period in the
+                            // aggregators
+                            final Optional<SamplingUnit> maxUnit = metric.getAggregators().stream()
+                                    .filter(agg -> agg.getAlignSampling().orElse(Boolean.FALSE)) // Filter out non-sampling aligned
+                                    .map(Aggregator::getSampling)
+                                    .map(sampling -> sampling.map(Sampling::getUnit).orElse(SamplingUnit.MILLISECONDS))
+                                    .min(SamplingUnit::compareTo);
+
+                            // No aggregators are sampling aligned so skip as rollups are always aligned
+                            if (maxUnit.isPresent()) {
+
+                                final TreeMap<SamplingUnit, String> orderedRollups = new TreeMap<>();
+                                rollupMetrics.forEach(name -> {
+                                    final Optional<SamplingUnit> rollupUnit =
+                                            rollupSuffixToSamplingUnit(name.substring(metricName.length() + 1));
+                                    rollupUnit.ifPresent(samplingUnit -> orderedRollups.put(samplingUnit, name));
+                                });
+
+                                final Map.Entry<SamplingUnit, String> floorEntry = orderedRollups.floorEntry(maxUnit.get());
+                                metrics.incrementCounter("kairosService/useRollups/noMatchingRollup", floorEntry != null ? 1 : 0);
+                                final String rollupName = floorEntry != null ? floorEntry.getValue() : metricName;
+                                final Metric.Builder metricBuilder = Metric.Builder.<Metric, Metric.Builder>clone(metric)
+                                        .setName(rollupName);
+
+                                return metricBuilder.build();
+                            } else {
+                                metrics.incrementCounter("kairosService/useRollups/notEligible", 1);
+                            }
+
+                            return metric;
+                        }
                     }
-
-                    return metric;
-                }
-            }
-        })
-                .collect(ImmutableList.toImmutableList()));
-
-
-        return newQueryBuilder.build();
```

Comment:
I skimmed this and verified that it looked like you'd basically just indented all this as a side effect of switching to ThreadLocalBuilder. Let me know if you actually changed the logic inside.

Reply:
There were other formatting and naming changes to this block aside from the indentation; however, there were no logical changes.

Pro Tip: 
* Click the gear icon at the top of the review
* Check `Hide whitespace changes`